memote/support/helpers.py

```diff
@@ -389,16 +451,181 @@ def open_boundaries(model):
+    compartment_id : string
+        Model specific compartment identifier.
 
     Returns
     -------
-    cobra.Model
-        A cobra model with all boundary reactions opened.
+    list
+        List of metabolites belonging to a given compartment.
 
     """
-    for exchange in model.exchanges:
-        exchange.bounds = (-1000, 1000)
+    return [met for
+            met in model.metabolites if
+            met.compartment == compartment_id]
+
+
+def largest_compartment_id_met(model):
+    """
+    Return the ID of the compartment with the most metabolites.
+
+    Parameters
+    ----------
+    model : cobra.Model
+        A cobrapy metabolic model
+
+    Returns
+    -------
+    string
+        Compartment ID of the compartment with the most metabolites.
+
+    """
+    size = {
+        compartment_id: len(metabolites_per_compartment(
+            model, compartment_id)) for
+        compartment_id in model.compartments.keys()
+    }
+    candidate = sorted(size, key=size.get, reverse=True)[0]
+    tie = size.copy()
+    tie.pop(candidate)
+    cand_tie = sorted(tie, key=tie.get, reverse=True)[0]
+    if size[candidate] not in tie.values():
+        return candidate
+    else:
+        raise RuntimeError("There is a tie for the largest compartment. "
+                           "Compartment {} and {} have equal amounts of "
+                           "metabolites.".format(candidate, cand_tie))
+
+
+def find_compartment_id_in_model(model, compartment_id):
+    """
+    Identify a model compartment by looking up names in COMPARTMENT_SHORTLIST.
+
+    Parameters
+    ----------
+    model : cobra.Model
+        A cobrapy metabolic model
+    compartment_id : string
+        Memote internal compartment identifier used to access compartment name
+        shortlist to look up potential compartment names.
+
+    Returns
+    -------
+    string
+        Compartment identifier in the model corresponding to compartment_id
+
+    """
+    if compartment_id not in COMPARTMENT_SHORTLIST.keys():
+        raise RuntimeError("{} is not in the COMPARTMENT_SHORTLIST! Make sure "
+                           "you typed the ID correctly, if yes, update the "
+                           "shortlist manually.".format(compartment_id))
+
+    if len(model.compartments) == 0:
+        raise RuntimeError(
+            "It was not possible to identify the "
+            "compartment {}, since the "
+            "model has no compartments at "
+            "all.".format(COMPARTMENT_SHORTLIST[compartment_id][0])
+        )
+
+    if compartment_id in model.compartments.keys():
+        return compartment_id
+
+    for name in COMPARTMENT_SHORTLIST[compartment_id]:
+        for c_id, c_name in model.compartments.items():
+            if c_name.lower() == name:
+                return c_id
+
+    if compartment_id == 'c':
+        return largest_compartment_id_met(model)
+
+
+def find_met_in_model(model, mnx_id, compartment_id=None):
```

Comment:
I think that letting this function raise a `KeyError` instead is more idiomatic.